src/Rotations.jl

```diff
@@ -547,19 +553,20 @@ function normal_rotation_distribution(M::Rotations{N}, x, σ::Real) where N
 end
 
 @doc doc"""
-    mean(M::Rotations, x::AbstractVector, w::AbstractWeights; shuffle_rng=nothing, kwargs...)
+    mean(M::Rotations, x::AbstractVector[, w::AbstractWeights]; shuffle_rng=nothing, kwargs...)
 
-Compute the Riemannian mean of `x` using [`GeodesicInterpolationMethod`](@ref).
-If any `x` are not within $\frac{\pi}{2 \sqrt 2}$ of the estimated mean, then
-the estimate is used as the initial guess for the gradient method.
+Compute the Riemannian [`mean`](@ref mean(M::Manifold, args...)) of `x` using
+[`GeodesicInterpolation`](@ref). If any `x` are not within
+$\frac{\pi}{2 \sqrt 2}$ of the estimated mean, then the estimate is used to
+initialize mean computation using the [`GradientDescent`](@ref).
 """
 mean(::Rotations, args...)
 
-function mean!(M::Rotations, y, x::AbstractVector, w::AbstractWeights; shuffle_rng = nothing, kwargs...)
-    mean!(M, y, x, w, GeodesicInterpolationMethod(); shuffle_rng = shuffle_rng, kwargs...)
+function mean!(M::Rotations, y, x::AbstractVector, w::AbstractVector; shuffle_rng = nothing, kwargs...)
+    mean!(M, y, x, w, GeodesicInterpolation(); shuffle_rng = shuffle_rng, kwargs...)
     for i in eachindex(x)
         @inbounds if distance(M, y, x[i]) ≥ π/2/√2
-            return mean!(M, y, x, w, GradientMethod(); x0 = y, kwargs...)
+            return mean!(M, y, x, w, GradientDescent(); x0 = y, kwargs...)
         end
     end
     return y
```

Comment:
For me this implementation looks very similar to the Grassmann one, just the injectivity_radius is different (within the if) – could we implement that in general and just set the default method for rotaition and Grassmann to this mix of GeodesicInterpolation and GradientDescent? This makes these lines to be just one line and we have just one implementation of this method.